content-api/content-actors/src/main/scala/org/sunbird/content/dial/DIALManager.scala

```diff
@@ -116,11 +121,60 @@ object DIALManager {
 		}).flatMap(f => f)
 	}
 
-	//TODO: Complete the implementation
-	def linkCollection(objectId: String, requestMap: Map[String, List[String]], getContext: util.Map[String, AnyRef])(implicit oec: OntologyEngineContext, ec: ExecutionContext): Future[Response] = {
-		Future {
-			ResponseHandler.OK()
-		}
+	def linkCollection(objectId: String, requestMap: Map[String, List[String]], reqContext: util.Map[String, AnyRef])(implicit oec: OntologyEngineContext, ec: ExecutionContext): Future[Response] = {
+		val request = new Request()
+		request.setContext(reqContext)
+		request.getContext.put(ContentConstants.SCHEMA_NAME, ContentConstants.COLLECTION_SCHEMA_NAME)
+		request.getContext.put(ContentConstants.VERSION, ContentConstants.SCHEMA_VERSION)
+		request.put(ContentConstants.ROOT_ID, objectId)
+		request.put(ContentConstants.MODE, "edit")
+
+		val req = new Request(request)
+		req.put("identifier", request.get("rootId").asInstanceOf[String])
+		req.put("mode", request.get("mode").asInstanceOf[String])
```

Comment:
Step repeated because above request just gives context information to new request. other request attributes do not get carried. I have updated constants for remaining string literals too.